components/style/properties/properties.mako.rs

```diff
@@ -1094,17 +1094,31 @@ impl PropertyDeclaration {
     /// Given a property declaration, return the property declaration id.
     pub fn id(&self) -> PropertyDeclarationId {
         match *self {
+            PropertyDeclaration::Custom(ref name, _) => {
+                return PropertyDeclarationId::Custom(name)
+            }
+            PropertyDeclaration::CSSWideKeyword(id, _) |
+            PropertyDeclaration::WithVariables(id, _) => {
+                return PropertyDeclarationId::Longhand(id)
+            }
+            _ => {}
+        }
+        let longhand_id = match *self {
             % for property in data.longhands:
                 PropertyDeclaration::${property.camel_case}(..) => {
-                    PropertyDeclarationId::Longhand(LonghandId::${property.camel_case})
+                    LonghandId::${property.camel_case}
                 }
             % endfor
-            PropertyDeclaration::CSSWideKeyword(id, _) => PropertyDeclarationId::Longhand(id),
-            PropertyDeclaration::WithVariables(id, _) => PropertyDeclarationId::Longhand(id),
-            PropertyDeclaration::Custom(ref name, _) => {
-                PropertyDeclarationId::Custom(name)
+            PropertyDeclaration::CSSWideKeyword(..) |
+            PropertyDeclaration::WithVariables(..) |
+            PropertyDeclaration::Custom(..) => {
+                // This value is never used, but having an expression of the same "shape"
+                // as for other variants helps the optimizer compile this `match` expression
+                // to a lookup table.
+                LonghandId::BackgroundColor
```

Comment:
Though I guess this optimization is in MIR. Not sure whether debug! stuff has been compiled out by then.

Anyway, I'd prefer it if we can do it easily, but otherwise don't spend too much time on it.